src/foam/mlang/mlangJava.js

```diff
@@ -323,6 +323,20 @@ foam.CLASS({
 });
 
 
+foam.CLASS({
+  refines: 'foam.mlang.predicate.Has',
+
+  methods: [
+    {
+      name: 'f',
+      javaCode: 'Object value = getArg1().f(obj); '+
+        'return ! (value == null || '+
```

Comment:
Are you sure this compiles?

I don't believe isEmpty() is a method on all Objects in java.

I also don't think we should compare the output of toString() as that may require building strings we don't need which is slow.